### PR TITLE
Added Optimizely Config in Typescript

### PR DESCRIPTION
**Pull request by fayyazarshad**
## Summary
Added Optimizely Config in Typescript

## TestPlan
Don't have any specific testplan but react sdk should work for OptimizelyConfig

### CODE REVIEW

File: packages/optimizely-sdk/lib/index.d.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { OptimizelyVariable } from "@optimizely/optimizely-sdk"
```
Comment (msohailhussain) [inline packages/optimizely-sdk/lib/index.d.ts]: @fayyazarshad this should be below header. Change header year.

File: packages/optimizely-sdk/lib/index.d.ts [hunk 2]
```diff
@@ -214,6 +217,65 @@ declare module "@optimizely/optimizely-sdk" {
...
+    id: string;
+    key: string;
+    type: string;
+    value: string;
+  }
+  
+  export interface OptimizelyVariation {
+    id: string;
+    key: string;
+    variablesMap: {
```
Comment (msohailhussain) [inline packages/optimizely-sdk/lib/index.d.ts]: featureEnabled?

File: packages/optimizely-sdk/lib/index.d.ts [hunk 3]
```diff
@@ -115,6 +116,7 @@ declare module "@optimizely/optimizely-sdk" {
       timeout?: number;
     }): Promise<{ success: boolean; reason?: string }>;
     close(): void;
+    getOptimizelyConfig(): OptimizelyConfig;
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/index.d.ts]:
```suggestion
    getOptimizelyConfig(): OptimizelyConfig | null;
```

File: packages/optimizely-sdk/lib/index.d.ts [hunk 4]
```diff
@@ -111,10 +111,11 @@ declare module "@optimizely/optimizely-sdk" {
       userId: string,
       attributes?: UserAttributes
     ): string | null;
+    getOptimizelyConfig(): OptimizelyConfig;
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/index.d.ts]:
```suggestion
    getOptimizelyConfig(): OptimizelyConfig | null;
```

**Pull request merged**